cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/batch/mapreduce/MapReduceBatchContext.java

```diff
@@ -66,6 +67,7 @@ public MapReduceBatchContext(MapReduceContext context, PipelineRuntime pipelineR
     this.inputNames = new HashSet<>();
     this.isPreviewEnabled = context.getDataTracer(stageSpec.getName()).isEnabled();
     this.connectorDatasets = Collections.unmodifiableSet(connectorDatasets);
+    this.maxPreviewRecordsLimit = context.getDataTracer(stageSpec.getName()).getMaximumTracedRecords();
```

Comment:
As mentioned in the comment above, no need to keep `maxPreviewRecordsLimit` as member of this class, instead create new stage spec which is copy of stageSpec parameter with number of preview records set appropriately.